Source/Caches/HACacheKeyStates.swift

```diff
@@ -0,0 +1,65 @@
+import Foundation
+
+/// Key for the cache
+internal struct HACacheKeyStates: HACacheKey {
+    static func create(connection: HAConnection) -> HACache<HACachedStates> {
+        .init(
+            connection: connection,
+            subscribe:
+            .init(
+                subscription: .subscribeEntities(),
+                transform: { info in
+                    .replace(processUpdates(info: info))
+                }
+            )
+        )
+    }
+
+    /// Process updates from the compressed state to HAEntity
+    /// - Parameter info: The compressed state update and the current cached states
+    /// - Returns: HAEntity cached states
+    /// Logic from: https://github.com/home-assistant/home-assistant-js-websocket/blob/master/lib/entities.ts
+    static func processUpdates(info: HACacheTransformInfo<HACompressedStatesUpdates, HACachedStates?>)
+        -> HACachedStates {
+        var states = info.current ?? .init(entities: [])
+
+        if let additions = info.incoming.add {
+            for (entityId, updates) in additions {
+                if var currentState = states[entityId] {
+                    currentState.update(from: updates)
+                    states[entityId] = currentState
+                } else {
+                    do {
+                        states[entityId] = try updates.asEntity(entityId: entityId)
+                    } catch {
+                        HAGlobal.log(.error, "Failed adding new entity: \(error)")
```

Comment:
You should still invoke this case in tests, if you can.

Reply:
I did a little workaround temporaraly implementing logs and then resetting it, let me know what you think